src/gt4py/next/iterator/type_system/type_synthesizer.py

```diff
@@ -270,82 +276,172 @@ def apply_lift(
     return apply_lift
 
 
-def _convert_as_fieldop_input_to_iterator(
-    domain: it_ts.DomainType, input_: ts.TypeSpec
-) -> it_ts.IteratorType:
-    # get the dimensions of all non-zero-dimensional field inputs and check they agree
+def _collect_and_check_dimensions(input_: ts.TypeSpec) -> list[common.Dimension]:
+    """
+    Extracts dimensions from non-zero-dimensional field inputs and ensures they match.
+    """
```

Comment:
Optional, feel free to ignore:
```suggestion
    """Extract dimensions from non-zero-dimensional field inputs and ensure they match."""
```